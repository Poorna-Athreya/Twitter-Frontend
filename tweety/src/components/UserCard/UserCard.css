.user-card{
    width:300px;
    height:auto;
    background-color: var(--card-bg);
    border: none;
    border-radius: 14px;
}
.user-card:hover{
    background-color: var(--card-bg-dark);
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
}
.user-handle{
    color: var(--accent-color-2);
}