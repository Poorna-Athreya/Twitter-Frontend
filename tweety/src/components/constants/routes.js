export const USERS_ROUTE = '/users';
export const TWEETS_ROUTE = '/tweets';
