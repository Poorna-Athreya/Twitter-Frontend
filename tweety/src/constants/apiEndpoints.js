export const BACKEND_URL = 'http://localhost:8000';
// export const TWEETS = '/users';
export const getUsers = {
  method: 'get',
  url: '/users',
};
export const getTweetsForUser = (userId) => ({
  method: 'get',
  url: `/tweets/${userId}`,
});
