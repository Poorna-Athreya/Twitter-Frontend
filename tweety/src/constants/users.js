export const ALL_USERS = [
  {
    id: 2244994945,
    name: 'Twitter Dev',
    handle: 'TwitterDev',
    location: 'Internet',
  },
  {
    id: 2244994946,
    name: 'Lorem Ipsum',
    handle: 'LoremIpsum',
    location: 'Internet',
  },
  {
    id: 2244994947,
    name: 'Jane Doe',
    handle: 'JaneDoe',
    location: 'Internet',
  },
  {
    id: 2244994947,
    name: 'Jane Doe',
    handle: 'JaneDoe',
    location: 'Internet',
  },
  {
    id: 2244994948,
    name: 'John Doe',
    handle: 'JohnDoe',
    location: 'Internet',
  },
];
export const USER_TWEETS = [
  {
    createdAt: 'Thu Apr 06 15:24:15 +0000 2017',
    text: 'Stguyg iugiuhniuhions',
    userId: 2244994948,
    id: 1,
  },
  {
    createdAt: 'Thu Apr 06 15:24:15 +0000 2017',
    text: 'Strings are used for sheet names. Integers are used in zero-indexed sheet positions',
    userId: 2244994948,
    id: 2,
  },
  {
    createdAt: 'Thu Apr 06 15:24:15 +0000 2017',
    text: 'Tweet 3 in zero-indexed sheet positions',
    userId: 2244994948,
    id: 3,
  },
  {
    createdAt: 'Thu Apr 06 19:24:15 +0000 2017',
    text: 'Strings are used for sheet names. Integers are used in zero-indexed sheet positions',
    userId: 2244994948,
    id: 4,
  },
];
// { text, userId; }
