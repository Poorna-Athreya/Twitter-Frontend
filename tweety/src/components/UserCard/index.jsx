import React from 'react';
import './UserCard.css';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import { TWEETS_ROUTE, USERS_ROUTE } from '../../constants/routes';

function UserCard({
  userName, userHandle, location, id,
}) {
  const navigate = useNavigate();
  return (
    <button aria-label="user-card-button" type="button" className="user-card" onClick={() => navigate(`${USERS_ROUTE}/${id}${TWEETS_ROUTE}`)}>
      <h1 className="user-name">{userName}</h1>
      <br />
      <div className="user-details">
        <p className="user-handle">
          @
          {userHandle}
        </p>
        <p className="user-location">{location}</p>
      </div>
    </button>
  );
}
UserCard.propTypes = {
  userName: PropTypes.string.isRequired,
  userHandle: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};
export default UserCard;
