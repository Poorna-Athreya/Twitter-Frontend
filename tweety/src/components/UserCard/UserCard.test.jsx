/* eslint-disable max-len */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import UserCard from './index';
import { TWEETS_ROUTE, USERS_ROUTE } from '../../constants/routes';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({ useNavigate: () => mockNavigate }));

describe('UserCard', () => {
  const mockUserCardName = 'Jane Doe';
  const mockUserCardHandle = 'JaneDoe';
  const mockUserCardLocation = 'Internet';
  const mockUserCardId = 2;
  beforeEach(() => {
    render(<UserCard userName={mockUserCardName} userHandle={mockUserCardHandle} location={mockUserCardLocation} id={mockUserCardId} />);
  });
  it('should have the required elements when UserCard is rendered', () => {
    expect(screen.getByTestId('userCardName')).toBeTruthy();
    expect(screen.getByText(mockUserCardName)).toBeTruthy();
    expect(screen.queryByTestId('userCardHandle')).toBeTruthy();
    expect(screen.getByText(`@${mockUserCardHandle}`)).toBeTruthy();
    expect(screen.queryByTestId('userCardLocation')).toBeTruthy();
    expect(screen.getByText(mockUserCardLocation)).toBeTruthy();
    expect(screen.queryByTestId('userCardButton')).toBeTruthy();
  });
  it('should call mockNavigate function when the button is clicked', () => {
    fireEvent.click(screen.getByTestId('userCardButton'));
    expect(mockNavigate).toHaveBeenCalledWith(`${USERS_ROUTE}/${mockUserCardId}${TWEETS_ROUTE}`);
  });
});
