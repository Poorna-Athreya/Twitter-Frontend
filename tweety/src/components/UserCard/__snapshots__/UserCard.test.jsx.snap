// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should take a snapshot for UserCard 1`] = `
<DocumentFragment>
  <button
    aria-label="user-card-button"
    class="user-card"
    data-testid="userCardButton"
    type="button"
  >
    <h1
      class="user-name"
      data-testid="userCardName"
    >
      Jane Doe
    </h1>
    <br />
    <div
      class="user-details"
    >
      <p
        class="user-handle"
        data-testid="userCardHandle"
      >
        @JaneDoe
      </p>
      <p
        class="user-location"
        data-testid="userCardLocation"
      >
        Internet
      </p>
    </div>
  </button>
</DocumentFragment>
`;
